src/tasks/task_list.hpp

```diff
@@ -70,23 +70,23 @@ class TaskList {
   }
 
   template <class F, class... Args>
-  TaskID AddTask(F func, TaskID &dep, Args &&... args) {
+  TaskID AddTask(F &&func, TaskID const &dep, Args &&... args) {
     TaskID id(tasks_added_ + 1);
     task_list_.push_back(
-        Task(id, dep, [=]() mutable -> TaskStatus { return func(args...); }));
+        Task(id, dep, [=, func{std::forward<F>(func)}]() mutable -> TaskStatus {
```

Comment:
What is this line doing? I've never seen a `std::forward` applied to a closure like this.

Reply:
Basically, if you pass a lambda directly to `AddTask` (i.e. rvalue-ref), it will move it instead of copy it. It will still copy those lambdas that are passed by lvalue ref. Again, originally this was to support adding tasks with move only objects in it, but then I ran into the `std::function` wall. Still, this is strictly "better" without removing anything that was previously supported.

Reply:
Huh. I see. That's subtle.